apps/extension/src/ui/domains/Staking/shared/useGetLatestBlockNumber.ts

```diff
@@ -0,0 +1,18 @@
+import { useQuery } from "@tanstack/react-query"
+
+import { ScaleApi } from "@ui/util/scaleApi"
+
+type GetLatestBlockNumber = {
+  sapi: ScaleApi | undefined | null
+  isEnabled: boolean
+}
+
+export const useGetLatestBlockNumber = ({ sapi, isEnabled }: GetLatestBlockNumber) => {
+  return useQuery({
+    queryKey: ["useGetLatestBlockNumber", sapi?.id],
```

Comment:
I got an error on this line when testing (no Taostats API key configured) - `Query data cannot be undefined. Please make sure to return a value other than undefined from your query function. Affected query key: ["useGetLatestBlockNumber",null]`